src/WebJobs.Script.WebHost/Routing/ScriptRouteHandler.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Authorization.Policy;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Azure.WebJobs.Script;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.WebHost.Features;
+using Microsoft.Azure.WebJobs.Script.WebHost.Security.Authorization;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.AspNetCore.Routing;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Http
+{
+    public class ScriptRouteHandler : IWebJobsRouteHandler
+    {
+        private readonly Func<ScriptHost> _scriptHostProvider;
+        private readonly ILoggerFactory _loggerFactory;
+
+        public ScriptRouteHandler(ILoggerFactory loggerFactory, Func<ScriptHost> scriptHostProvider)
+        {
+            _scriptHostProvider = scriptHostProvider;
+            _loggerFactory = loggerFactory;
+        }
+
+        public ScriptRouteHandler(ILoggerFactory loggerFactory, ScriptHost scriptHost)
+        {
+            _loggerFactory = loggerFactory;
+            _scriptHostProvider = () => scriptHost;
+        }
+
+        public async Task InvokeAsync(HttpContext context, string functionName)
+        {
+            IActionResult result = await GetResultAsync(context, functionName);
+
+            var actionContext = new ActionContext
+            {
+                HttpContext = context
+            };
+
+            await result.ExecuteResultAsync(actionContext);
+        }
+
+        private async Task<IActionResult> GetResultAsync(HttpContext context, string functionName)
+        {
+            ScriptHost host = _scriptHostProvider();
+
+            // TODO: FACAVAL This should be improved....
+            FunctionDescriptor descriptor = host.Functions.FirstOrDefault(f => string.Equals(f.Name, functionName));
```

Comment:
Where is all the logic we had before in FunctionsController that checked if a function was disabled, allowing it to be invoked if the request has admin auth, etc?